### PR TITLE
[FIX] Derivatives datasink handling of compression

### PR DESCRIPTION
**Pull request by mgxd**
not fully tested - adapted from @effigies  https://github.com/poldracklab/fmriprep/issues/1075#issuecomment-385057946

### CODE REVIEW

File: fmriprep/interfaces/bids.py [hunk 1]
```diff
@@ -420,3 +416,15 @@ def _splitext(fname):
         fname, ext2 = op.splitext(fname)
         ext = ext2 + ext
     return fname, ext
+
+
+def _copy_any(src, dst):
+    src_isgz = src.endswith('.gz')
+    dst_isgz = dst.endswith('.gz')
+    if src_isgz == dst_isgz:
+        copyfile(src, dst, use_hardlink=True)
```
Comment (effigies) [inline fmriprep/interfaces/bids.py]: Need `copy=True` in there, otherwise we might end up with symlinks.

**Comment by effigies** [comment]
Looks good, thanks. @oesteban do we want to add any sanity checks, such as making sure that all image files in the output can at least be `nibabel.load`ed?

Actually, maybe we should add `--low-mem` to one of the tests, so that the compression pathway actually gets hit.

**Comment by oesteban** [comment]
Adding `--low-mem` to one of the tests seems like a good idea :+1: .

I think the `nibabel.load` test can be done in a different PR. We probably want to check the hash of the outputs and force ourselves to update manually the hashes when they change (as it is done now for MRIQC).

**Comment by oesteban** [comment]
I've opened #1078. I'll merge this in since it is quite urgent.

**Pull request merged**